idl/group0.idl.hh

```diff
@@ -17,4 +17,11 @@ struct group0_peer_exchange {
     std::variant<std::monostate, service::group0_info, std::vector<raft::server_address>> info;
 };
 
+enum class group0_upgrade_state : uint8_t {
```

Comment:
This is the best place to document these states, their purpose and relationship.